activerecord/test/cases/calculations_test.rb

```diff
@@ -392,6 +392,17 @@ def test_distinct_count_with_group_by_and_order_and_limit
     assert_equal({ 6 => 2 }, Account.group(:firm_id).distinct.order("1 DESC").limit(1).count)
   end
 
+  def test_count_for_a_composite_primary_key_model
+    book = cpk_books(:cpk_great_author_first_book)
+    assert_equal(1, Cpk::Book.where(author_id: book.author_id, number: book.number).count)
+  end
+
+  def test_group_by_count_for_a_composite_primary_key_model
+    book = cpk_books(:cpk_great_author_first_book)
+    expected = { book.author_id => Cpk::Book.where(author_id: book.author_id).count }
+    assert_equal(expected, Cpk::Book.where(author_id: book.author_id).group(:author_id).count)
+  end
+
```

Comment:
>Do we need any SQL assertions here?

Not in these cases since here we are not using the primary key in the SQL query as the `count` one generates roughly `select count(*) from cpk_books where author_id=1 and number = 1` (even though columns match the PK they are not automatically derived) and the `group` one generates
`select count(*) from cpk_books where author_id = 1 group by author_id` so we actually expect the `primary_key` not to be used and our main expectation here is that `primary_key` being an Array doesn't interfere with these flows